src/overlays/actors/ovl_En_Dodongo/z_en_dodongo.c

```diff
@@ -136,97 +279,837 @@ static DamageTable D_808792D4 = {
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877494.s")
+    if (this->timer != 0) {
+        this->timer--;
+        if (this->timer == 0) {
+            if (this->actor.params == 0) {
+                Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.world.pos, 0x90);
+            } else {
+                Item_DropCollectible(globalCtx, &this->actor.world.pos, ITEM00_RUPEE_PURPLE);
+            }
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877500.s")
+void EnDodongo_UpdateDamage(EnDodongo* this, GlobalContext* globalCtx) {
+    Vec3f sp3C;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_808777A8.s")
+    if (this->collider2.base.acFlags & AC_HIT) {
+        this->collider2.base.acFlags &= ~AC_HIT;
+        this->collider1.base.acFlags &= ~AC_HIT;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_8087784C.s")
+        for (i = 0; i < ARRAY_COUNT(this->collider2Elements); i++) {
+            if (this->collider2.elements[i].info.bumperFlags & BUMP_HIT) {
+                break;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877D50.s")
+        if ((i != ARRAY_COUNT(this->collider2Elements)) &&
+            ((this->unk_300 != 10) || !(this->collider2.elements[i].info.acHitInfo->toucher.dmgFlags & 0xDB0B3))) {
+            func_80876D28(this, globalCtx);
+            Math_Vec3s_ToVec3f(&sp3C, &this->collider2.elements[i].info.bumper.hitPos);
+            if (this->actor.colChkInfo.damageEffect == 0xF) {
+                CollisionCheck_BlueBlood(globalCtx, NULL, &sp3C);
+                EffectSsHitMark_SpawnFixedScale(globalCtx, 0, &sp3C);
+            } else if (this->actor.colChkInfo.damageEffect != 14) {
+                EffectSsHitMark_SpawnFixedScale(globalCtx, 3, &sp3C);
+                CollisionCheck_SpawnShieldParticlesMetalSound(globalCtx, &sp3C, &this->actor.projectedPos);
+            }
+        }
+    } else if (this->collider1.base.acFlags & AC_HIT) {
+        this->collider2.base.acFlags &= ~AC_HIT;
+        this->collider1.base.acFlags &= ~AC_HIT;
+        func_800BE2B8(&this->actor, &this->collider1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877D90.s")
+        for (i = 0; i < ARRAY_COUNT(this->collider1Elements); i++) {
+            if (this->collider1.elements[i].info.bumperFlags & BUMP_HIT) {
+                break;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Dodongo/func_80877DE0.s")
+        if ((i != ARRAY_COUNT(this->collider1Elements)) &&
+            ((this->unk_300 != 10) || !(this->collider1.elements[i].info.acHitInfo->toucher.dmgFlags & 0xDB0B3))) {
+            func_80876D28(this, globalCtx);
+            if (this->actor.colChkInfo.damageEffect != 15) {
```

Comment:
```suggestion
            if (this->actor.colChkInfo.damageEffect != 0xF) {
```
Keeping it consistent with what you have above